components/nodemanager-service/pgdb/nodes.go

```diff
@@ -370,7 +369,15 @@ func (db *DB) BulkAddNodes(inNodes []*nodes.Node) (nodeIDs []string, err error)
 		if err != nil {
 			return nodeIDs, errors.Wrap(err, "BulkAddNodes unable to translate node to db struct")
 		}
-		for _, host := range inNode.GetTargetConfig().GetHosts() {
+
+		hosts := inNode.GetTargetConfig().GetHosts()
+		if inNode.GetTargetConfig().Host != "" {
+			hosts = append(hosts, inNode.GetTargetConfig().Host)
+		}
+		if len(hosts) < 1 {
```

Comment:
@vjeffrey had this feedback here, but got lost we I opened the PR:

_👍 code makes sense, but i wonder...should we be strict and reject the call (at the api level) is a host field is non empty on a bulk nodes call, b/c they should be sticking everything in hosts?_

Reply:
A host can be passed in the BulkAddNodes payload at the moment via:

```
target_config:
  host: string
```
and/or
```
target_config:
  hosts: []string
```
My change accepts both and only errors out if none are provided.

I could be convinced as well to only accept `target_config.hosts` in the BulkAddNodes call. This wouldn't break existing scripts as we ignored `target_config.host` for this call.

Reply:
So expanding on that bit above, if we weren't sharing the message definition between calls, would we have ever provided users with the ability to add hosts via a singular string host field and an array of strings hosts field for an endpoint used for bulk node add?  i would argue probably not - which is why I lean towards going with the strict approach

Reply:
After sleeping on it, to avoid muddying the waters, it makes sense to only process `target_config.hosts` for the bulk endpoint. I pushed another commit where the API errors out if `target_config.host` is used for the bulk endpoint.